### PR TITLE
Alerting - add Event as an Alert Recipient type

### PR DESCRIPTION
**Pull request by bananaumai**
# About

This PR attempts to support MODE Event as an Alert Recipient Type.
The API client can specify the "event type" of MODE Events propagated when Alerts are invoked.

# Note

The backend supporting this feature will be released in the near future.

### CODE REVIEW

**Review by bill-mode — APPROVED** [review]

**Comment by bill-mode** [comment]
@bananaumai, I have some changes I need to merge first. Please wait before merging this to master. I will let you know after I merged my branch. Thanks.

File: mode-apis-base/src/alertingModels.ts
Comment (bill-mode) [inline mode-apis-base/src/alertingModels.ts:189]: Can this be any string? Or is there a predefined list of possible string the user can choose from?

Comment (bananaumai) [reply mode-apis-base/src/alertingModels.ts:189]: As of now, we don't have a way to list possible event types.
So, it's okay to leave it just an arbitrary string.

In the backend, eventType is validated by the following regex.

https://github.com/moderepo/platform/blob/86f4f1068e050961419a4e3ff8e94047af6f53be/components/pkg/strutil/strutil.go#L22

Comment (bill-mode) [reply mode-apis-base/src/alertingModels.ts:189]: Ok, when I implement the front end, I will make sure to use the same regex

**Comment by bill-mode** [comment]
@bananaumai, you can merge this to main when you are ready.

**Pull request merged**
